contracts/receivables/BaseCreditPoolReceivable.sol

```diff
@@ -0,0 +1,235 @@
+     * @dev Makes a payment for a receivable.
+     * Only the owner of the token can make a payment.
+     * The payment method for the receivable must be Payable.
+     * The receivable must not already be paid in full.
+     * Calls the `makePayment` function of the corresponding `BaseCreditPool` contract.
+     * @param tokenId The ID of the receivable token.
+     * @param paymentAmount The amount of payment being made
+     *   (denoted in the paymentToken of the receivable).
+     */
+    function makePayment(uint256 tokenId, uint96 paymentAmount) public {
+        require(msg.sender == ownerOf(tokenId), "Caller is not token owner");
+        ReceivableInfo storage receivableInfo = receivableInfoMapping[tokenId];
+        require(
+            receivableInfo.paymentMethod == PaymentMethod.Payable,
+            "Unsupported payment method for receivable"
+        );
+        require(
+            receivableInfo.balance < receivableInfo.receivableAmount,
+            "Receivable already paid"
+        );
+
+        (uint256 amountPaid, ) = receivableInfo.baseCreditPool.makePayment(
+            ownerOf(tokenId),
+            uint256(paymentAmount)
+        );
+
+        require(amountPaid > 0, "makePayment failed");
+
+        receivableInfo.balance += uint96(amountPaid);
+    }
+
+    /**
+     * @dev Gets the payment status of a receivable.
+     * Returns `Status.Paid` if the receivable has been paid in full.
+     * Returns `Status.PartiallyPaid` if the receivable has been paid partially.
+     * Returns `Status.Unpaid` if the receivable has not been paid at all.
+     * @param tokenId The ID of the receivable token.
+     * @return The payment status of the receivable.
+     */
+    function getPaymentStatus(uint256 tokenId) public view returns (Status) {
+        ReceivableInfo storage receivableInfo = receivableInfoMapping[tokenId];
+        if (receivableInfo.balance == receivableInfo.receivableAmount) {
+            return Status.Paid;
+        } else if (receivableInfo.balance > 0) {
+            return Status.PartiallyPaid;
+        } else {
+            return Status.Unpaid;
+        }
+    }
+
+    // The following functions are overrides required by Solidity.
+    function _beforeTokenTransfer(
+        address from,
+        address to,
+        uint256 tokenId
+    ) internal override(ERC721, ERC721Enumerable) {
+        super._beforeTokenTransfer(from, to, tokenId);
+    }
+
+    function _burn(uint256 tokenId) internal override(ERC721, ERC721URIStorage) {
+        super._burn(tokenId);
+    }
+
+    function tokenURI(uint256 tokenId)
+        public
+        view
+        override(ERC721, ERC721URIStorage)
+        returns (string memory)
+    {
+        return super.tokenURI(tokenId);
+    }
+
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        override(ERC721, ERC721Enumerable, AccessControl)
+        returns (bool)
+    {
+        return super.supportsInterface(interfaceId);
```

Comment:
Here super is AccessControl only, ERC721Enumerable is missing. Need to call them together

```
return
            ERC721Enumerable.supportsInterface(interfaceId) ||
            AccessControl.supportsInterface(interfaceId);
```